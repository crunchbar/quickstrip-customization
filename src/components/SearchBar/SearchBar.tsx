import React from 'react';
import {createStyles, makeStyles, Theme} from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import InputAdornment from '@material-ui/core/InputAdornment';
import IconButton from '@material-ui/core/IconButton';
import CancelIcon from '@material-ui/icons/Cancel';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
    },
    inputRoot: {
      flexWrap: 'wrap'
    },
    inputInput: {
      width: 'auto',
      flexGrow: 1
    },
  })
);

export interface SearchBarProps {
  label: string;
  placeholder: string;
  value: string;
  onChange: (value: string) => void;
  className?: string;
  inputClassName?: string;
  id?: string;
}

const SearchBar: React.FC<SearchBarProps> = ({
  label,
  placeholder,
  value,
  onChange,
  className,
  inputClassName,
  id,
}) => {
  const classes = useStyles();
  const textFieldProps = {
    id,
    fullWidth: true,
    label,
    value,
    placeholder,
    InputLabelProps: { shrink: true },
    InputProps: {
      classes: {
        root: classes.inputRoot,
        input: classes.inputInput,
      },
      endAdornment: value ? (
        <InputAdornment position="end">
          <IconButton
            edge="end"
            aria-label="Clear Search"
            onClick={() => onChange('')}
          >
            <CancelIcon />
          </IconButton>
        </InputAdornment>
      ) : undefined,
    },
    inputProps: {
      className: inputClassName,
    },
    onChange: (e: any) => onChange(e.target.value),
  };
  return (
    <div className={`${classes.root} ${className ? className : ''}`}>
      <TextField {...textFieldProps} />
    </div>
  );
}

export default SearchBar;
