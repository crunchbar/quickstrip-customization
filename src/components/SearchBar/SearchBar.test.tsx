import * as React from 'react';
import {shallow} from 'enzyme';
import SearchBar, {SearchBarProps} from './SearchBar';

const props: SearchBarProps = {
  label: 'label',
  placeholder: 'placeholder',
  value: '',
  onChange: () => {},
};

describe('<SearchBar />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<SearchBar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
