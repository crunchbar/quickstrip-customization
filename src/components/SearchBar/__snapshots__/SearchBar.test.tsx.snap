// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> renders correctly 1`] = `
<div
  className="makeStyles-root-1 "
>
  <WithStyles(ForwardRef(TextField))
    InputLabelProps={
      Object {
        "shrink": true,
      }
    }
    InputProps={
      Object {
        "classes": Object {
          "input": "makeStyles-inputInput-3",
          "root": "makeStyles-inputRoot-2",
        },
        "endAdornment": undefined,
      }
    }
    fullWidth={true}
    inputProps={
      Object {
        "className": undefined,
      }
    }
    label="label"
    onChange={[Function]}
    placeholder="placeholder"
    value=""
  />
</div>
`;
